crates/ruff_graph/src/collector.rs

```diff
@@ -32,6 +32,28 @@ impl<'a> Collector<'a> {
 }
 
 impl<'ast> SourceOrderVisitor<'ast> for Collector<'_> {
+    fn enter_node(&mut self, node: AnyNodeRef<'ast>) -> TraversalSignal {
+        // If string detection is enabled, we have to visit everything. Otherwise, we should only
+        // visit compounds statements, which can contain import statements.
+        if self.string_imports
+            || matches!(
+                node,
+                AnyNodeRef::ModModule(_)
+                    | AnyNodeRef::StmtFunctionDef(_)
+                    | AnyNodeRef::StmtClassDef(_)
+                    | AnyNodeRef::StmtWhile(_)
+                    | AnyNodeRef::StmtFor(_)
+                    | AnyNodeRef::StmtWith(_)
+                    | AnyNodeRef::StmtIf(_)
+                    | AnyNodeRef::StmtTry(_)
+            )
```

Comment:
Won't this skip over `StmtImportFrom` imports if `self.string_imports` is `false`?a